lib/TeleIrc.js

```diff
@@ -188,6 +188,35 @@ class TeleIrc {
   initStage3_initBots(ircbot, tgbot) {
     this.ircbot = ircbot;
     this.tgbot = tgbot;
+    // This is super rudimentary error handling. If the Telegram bot
+    // throws an error, the library we use simply logs it and appears
+    // to hang. This may be a bug in that library (see
+    // https://github.com/yagop/node-telegram-bot-api/issues/657).
+    // This code forces teleirc to fully crash when it encounters a polling
+    // error so that its parent process/container can restart it.
+    // It may be possible to handle this in a nicer way, but issues
+    // open on the bot library indicate that things may not be working right
+    // as of October 2018.
+    this.tgbot.on('polling_error', (error) => {
+        console.log("Fatal Telegram polling error: " + error.code);
+        console.log(error.stack);
+        process.exit(1);
+    });
+    this.tgbot.on('webhook_error', (error) => {
+        console.log("Fatal Telegram webhook error: " + error.code);
+        console.log(error.stack);
+        process.exit(1);
+    });
+
+    // Verifies that we can, in fact, reach the Telegram API and
+// are not being rate limited.
+    let getmePromise = this.tgbot.getMe();
```

Comment:
Could use async/await here... though since this is only one promise, and not a giant cascade of them (like the imgur stuff was), I'm okay with leaving it as is.